[FLINK-36761][docs] Add document about how to use SQL Gateway and SQL Client to deploy script (#25784)
apache/flink: docs/content/docs/dev/table/sqlClient.md

<span class="label label-danger">Attention</span> Compared to the interactive mode, SQL Client will stop execution and exits when there are errors.

> ### Deploy SQL Files in an Application Cluster

Reviewer: Deploy SQL Files to an Application Cluster  Or Deploy SQL Files in Application Mode  ?

SQL Client also supports to deploy a SQL script file to an application cluster with the `-f` option if you specify the deployment target in the config.yaml or startup options.

> Here is an example to deploy script file in an application cluster.

Reviewer 2: Application Cluster is sometimes capitalised and sometimes not - I suggest using the capitalised form consistently

```bash 
./bin/sql-client.sh -f oss://path/to/script.sql \
      -Dexecution.target=kubernetes-application \
      -Dkubernetes.cluster-id=${CLUSTER_ID} \
      -Dkubernetes.container.image.ref=${FLINK_IMAGE_NAME}'
```

> After execution, SQL Client print the cluster id on the terminal. The script can contain any statement that is supported by Flink. But Application cluster only supports one job, please refer to the 

Reviewer: ```suggestion
After execution, SQL Client will print the cluster id on the terminal. The script can contain any statement supported by Flink. But Application cluster only supports one job, please refer to the 
```

> After execution, SQL Client print the cluster id on the terminal. The script can contain any statement that is supported by Flink. But Application cluster only supports one job, please refer to the 

Reviewer: > any statement that is supported by Flink. But Application cluster only supports one job

Are you sure this limitation is correct? If the script contains multiple DML statements, shouldn’t it be possible to submit multiple jobs in one application cluster ?

Author: I just mean the new feature supports to run any kinds of statement, but it doesn't mean the combination of the sql statements. 

If the cluster doesn't enable ha mode, the cluster is able to run multiple jobs.

[Application Mode]({{< ref "docs/deployment/overview#application-mode" >}}) for the limitations.

> <span class="label label-danger">Attention</span> When deploying a script to the cluster, SQL Client only supports to run with `--jars` startup option, other options, e.g. `--init` 

Reviewer 2: not:
to run -> running
